docs/transactions.txt

```diff
@@ -9,71 +9,148 @@ Transactions
+that ensures that all changes within a transaction are either applied or
+discarded is called **atomicity**.
+
+MongoDB performs write operations on single documents atomically. If you
+need atomicity in write operations on multiple documents or data consistency
+across multiple documents for your operations, run them in a multi-document
+transaction.
+
+Multi-document transactions are **ACID compliant** because MongoDB
+guarantees that the data in your transaction operations remains consistent,
+even if the driver encounters unexpected errors.
+
+Learn how to perform transactions in the following sections of this guide:
+
+- :ref:`laravel-transaction-requirements`
+- :ref:`laravel-transaction-callback`
+- :ref:`laravel-transaction-commit`
+- :ref:`laravel-transaction-rollback`
+
+To learn more about transactions in MongoDB, see :manual:`Transactions </core/transactions/>`
+in the {+server-docs-name+}.
+
+.. _laravel-transaction-requirements:
+
+Requirements and Limitations
+----------------------------
+
+To perform transactions in MongoDB, you must be using the following MongoDB
+version and topology:
 
 - MongoDB version 4.0 or later
 - A replica set deployment or sharded cluster
 
-You can find more information :manual:`in the MongoDB docs </core/transactions/>`
+MongoDB server and {+odm-short+} have the following limitations:
 
-.. code-block:: php
+- In MongoDB versions 4.2 and earlier, write operations performed within a
+  transaction must be on existing collections. In MongoDB versions 4.4 and
+  later, the server automatically creates collections as necessary when
+  you perform write operations in a transaction. To learn more about this
+  limitation, see :manual:`Create Collections and Indexes in a Transaction </core/transactions/#create-collections-and-indexes-in-a-transaction>`
+  in the {+server-docs-name+}.
+- MongoDB does not support nested transactions. If you attempt to start a
+  transaction within another one, the extension raises a ``RuntimeException``.
+  To learn more about this limitation, see :manual:`Transactions and Sessions </core/transactions/#transactions-and-sessions>`
+  in the {+server-docs-name+}.
+- The {+odm-long+} does not support the database testing traits
+  ``Illuminate\Foundation\Testing\DatabaseTransactions`` and ``Illuminate\Foundation\Testing\RefreshDatabase``.
+  As a workaround, you can create migrations with the ``Illuminate\Foundation\Testing\DatabaseMigrations``
+  trait to reset the database after each test.
 
-   DB::transaction(function () {
-       User::create(['name' => 'john', 'age' => 19, 'title' => 'admin', 'email' => 'john@example.com']);
-       DB::collection('users')->where('name', 'john')->update(['age' => 20]);
-       DB::collection('users')->where('name', 'john')->delete();
-   });
+.. _laravel-transaction-callback:
 
-.. code-block:: php
+Run a Transaction in a Callback
+-------------------------------
+
+This section shows how you can run a transaction as a callback.
```

Comment:
S: since you use the "in a callback" phrasing in the section header:
```suggestion
This section shows how you can run a transaction in a callback.
```